File: Data Structures & Algorithms/daily-temperatures/submission-3.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        for i in range(n - 2, -1, -1):
            j = i + 1
            while j < n and temperatures[j] <= temperatures[i]:
                j += res[j] if res[j] > 0 else n
            if j < n:
                res[i] = j - i
        return res
            

File: Data Structures & Algorithms/non-overlapping-intervals/submission-3.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (i1, i2) -> i1[0] - i2[0]);
        int res = 0;
        int prevEnd = intervals[0][1];
        for (int i = 1; i < intervals.length; i++) {
            int[] currInterval = intervals[i];
            if (prevEnd > currInterval[0]) {
                prevEnd = Math.min(prevEnd, currInterval[1]);
                ++res;
            } else {
                prevEnd = currInterval[1];
            }

        }

        return res;
    }
}
